core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -50,12 +50,16 @@ oppia.factory('explorationData', [
     var explorationData = {
       explorationId: explorationId,
       autosaveChangeList: function(changeList, successCallback, errorCallback) {
+        // First save locally to be retrieved later if save is unsuccessful.
+        localSaveService.saveExplorationDraft(explorationId, changeList,
+          draftChangeListId);
         $http.put(explorationDraftAutosaveUrl, {
           change_list: changeList,
           version: explorationData.data.version
         }).then(function(response) {
+          draftChangeListId = response.data.draft_change_list_id;
+          localSaveService.removeExplorationDraft(explorationId);
           if (successCallback) {
-            draftChangeListId = response.data.draft_change_list_id;
             successCallback(response);
           }
         }, function() {
```

Comment:
I'm not entirely sure what you mean. The draftChangeListId is only sent from the back-end to the front-end (on autosaves and loads) so that shouldn't be possible. In the scenario you described the draftChangeListId on the locally saved draft would differ from that on the back-end so the it would be discarded (preferably after showing a modal displaying the changes which may be lost).  Does that address your concern or have I misunderstood? Thanks!